.github/workflows/build-and-publish-master.yml

```
@@ -1,25 +1,148 @@
-# Build and publish for each push  to branch master
+# Build and publish for each push to branch master
 
-name: GitHub Actions - Tagged release
+name: GitHub Actions - Publish plugin modules
 
 on:
   push:
     branches:
-      - 'main'
+      - 'master'
 
 jobs:
-  build-and-publish-master:
+  build-and-publish-master-publish-cordapp:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
 
+      - name: Verify Changed files
+        uses: tj-actions/changed-files@v35
+        id: verify-publish-cordapp-updated
+        with:
+          files: |
+            publish-cordapp-conventions/gradle.properties
+
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'corretto'
+          java-version: '8'
+          cache: 'gradle'
```

Worth adding this `step.if` check so we won't load cache and install java on any push to the master branch

_Ideally it would be better to separate `tj-actions/changed-files` steps into one job that others would depend on, but **it's not a priority at the moment**_
```suggestion
      - uses: actions/setup-java@v3
        if: steps.verify-publish-cordapp-updated.outputs.any_changed == 'true'
        with:
          distribution: 'corretto'
          java-version: '8'
          cache: 'gradle'
```